kivy/input/providers/gpioinput.py

```diff
@@ -0,0 +1,270 @@
+
+                if ev_type == EV_SYN:
+                    if ev_code == SYN_REPORT:
+                        if not self._pending_axis:
+                            return
+
+                        ev_props = {'push_attrs': list(self._pending_axis),
+                                    'is_touch': False,
+                                    'is_rotary': True}
+                        for ax_coord in self._pending_axis:
+                            ev_props[ax_coord] = self._axis_info[ax_coord]
+                            self._axis_info[ax_coord] = 0
+
+                        Logger.trace(
+                            "GPIOInput:(%0.3f) <%s>[<EV_SYN>,<SYN_REPORT>] %s"
+                            % (tv_msec / 1000, self.device, ev_props))
+
+                        self.dispatch_queue.append(
+                            GPIOEvent(self.device,
+                                      'GPIOEvent_%d' % tv_msec,
+                                      ev_props))
+                        self._pending_axis.clear()
+
+                elif ev_type == EV_REL:
+                    ax_coord = AXIS_COORDS[ev_code]
+                    self._axis_info[ax_coord] += ev_value
+                    self._pending_axis.add(ax_coord)
+
+                    Logger.trace(
+                        "GPIOInput:(%0.3f)<%s>[<EV_REL>,AXIS %d(%s),VALUE %d]"
+                        % (tv_msec / 1000, self.device, ev_code, ax_coord,
+                           ev_value))
+
+            # open the input
+            fd = open(self.input_fn, 'rb')
+
+            # get the controller name (EVIOCGNAME)
+            self.device_name = str(fcntl.ioctl(fd,
+                                               EVIOCGNAME + (256 << 16),
+                                               " " * 256)
+                                   ).split('\x00')[0]
+
+            Logger.info('GPIOEvent: using <%s> as <%s>' %
+                        (self.device_name, self.device))
+
+            # read until the end
+            while fd:
+
+                data = fd.read(struct_input_event_sz)
+                if len(data) < struct_input_event_sz:
+                    break
+
+                # extract each event
+                for i in range(int(len(data) / struct_input_event_sz)):
+                    ev = data[i * struct_input_event_sz:]
+                    # extract timeval + event infos
+                    infos = struct.unpack('LLHHi', ev[:struct_input_event_sz])
+                    process_as_rotary(*infos)
+
+        def update(self, dispatch_fn):
+            # dispatch all events from threads
+
+            try:
+                while True:
+                    me = self.dispatch_queue.pop(0)
+                    Window.dispatch('on_motion', 'begin', me)
+                    Logger.trace('GPIOEvent: dispatching %r' % me)
+            except IndexError:
+                pass
+
+    MotionEventFactory.register('gpioinput', GPIOEventProvider)
```

Comment:
As far as I can see, only the `register` needs to be under `if 'KIVY_DOC' in os.environ`. Can you please place only that under the if statement. Eventually we will find a better solution than conditionally defining things based on `KIVY_DOC`, so it would be best if the minimally amount of things would be under it.